### DISCUSSION TITLE
Getting request data or url for a log RequestID ?

### DISCUSSION THREAD

**Discussion by LudovicLemarinel** [brefphp/laravel-bridge#138] [Q&A]
Hi,
I have a reccuring error in my logs, but I can't manage to find the corresponding URL that leads to this error : 

```
{
  "errorType": "Exception",
  "errorMessage": "The Lambda response cannot be encoded to JSON.\nThis error usually happens when you try to return binary content. If you are writing an HTTP application and you want to return a binary HTTP response (like an image, a PDF, etc.), please read this guide: https://bref.sh/docs/runtimes/http.html#binary-requests-and-responses\nHere is the original JSON error: 'Malformed UTF-8 characters, possibly incorrectly encoded'",
  "stack": [
    "#0 /var/task/vendor/bref/bref/src/Runtime/LambdaRuntime.php(181): Bref\\Runtime\\LambdaRuntime->postJson('http://127.0.0....', Array)",
    "#1 /var/task/vendor/bref/bref/src/Runtime/LambdaRuntime.php(91): Bref\\Runtime\\LambdaRuntime->sendResponse('a566dadc-f6e0-4...', Array)",
    "#2 /var/task/vendor/bref/bref/src/FunctionRuntime/Main.php(37): Bref\\Runtime\\LambdaRuntime->processNextEvent(Object(Bref\\LaravelBridge\\Http\\OctaneHandler))",
    "#3 /opt/bref/bootstrap.php(17): Bref\\FunctionRuntime\\Main::run()",
    "#4 {main}"
  ]
}
```

I understand that this error is because one of my routes returns some binary data that should be encoded, but my app has more than 200 routes, and I'm not willing to review them one after each ;)

Is there a way to get the corresponding url of the RequestId which is stored in CloudWatch ?
Or perhaps temporarly increase the verbose level of logs to track where does this error come from ?

Thanks ;)

**Comment by mnapoli** [comment]
You don't need to review your routes. Just follow the documentation that is mentioned in the error message and this will work for all routes. You don't have anything else to do, it should "just work" once you follow the steps.

**Reply by LudovicLemarinel** [reply]
Yes I've seen that page, but what I don't understand is that I don't return any binary data from my app !
So I'd like to know which route is throwing (garbage) data so I'll can add some logs to understand what is the problem ;)

**Reply by mnapoli** [reply]
OK, understood. So to answer your original questions: Bref does not log URLs or any application data. There is no option for that. If you want to correlate application data with this error you will have to log it yourself.

**Reply by LudovicLemarinel** [reply]
Thanks for your reply !
I've added a simple middleware that logs every request. I'll disable it as soon I'll have find which is the guilty controller ( since I suppose that storing big amounts of logs will probably invoiced from AWS !)
